ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+- `TargetBlockRate`: 2 (seconds)
+- `MinBlockGasCost`: 0 (gas)
+- `MaxBlockGasCost`: 1,000,000 (gas)
+- `BlockGasCostStep`: 200,000 (gas)
+
+`MinGasPrice` is equivalent to `M` in ACP-176 and will be used to set the minimum gas price for ACP-176. This is similar to `MinBaseFee` in old Subnet-EVM fee configuration, and roughly gives the same effect. Currently default value is `25 * 10^-18^` (25 nAVAX/Gwei). This default will be changed to the minimum possible denomination of the native EVM asset (1 Wei).
+
+`TargetGas` is equivalent to `T` (target gas consumed per second) in ACP-176 and will be used to set the target gas consumed per second for ACP-176.
+
+`MaxCapacityFactor` is equivalent to factor in `C` in ACP-176 and controls the maximum gas capacity (i.e block gas limit). This determines the `C` as `C = MaxCapacityFactor * T`. Currently default value is 10 as in C-Chain.
+
+`TimeToDouble` will be used to control the speed of the fee adjustment (`K`). This determines the `K` as `K = (RMult-1) * 1/ln2 * TimeToDouble`, where `RMult` is the factor in `R` which is defined as 2. Currently default value for `TimeToDouble` is 60 (seconds), making `K=~87*T` as in C-Chain.
+
+As a result parameters will be set as follows:
+
+| Parameter | Description | Default Configuration | Is Dynamic |
+| :--- | :--- | :--- | :--- |
+| T | target gas consumed per second | 2,000,000 | :white_check_mark: |
+| R | gas capacity added per second | 2*T | :x:
+| C | maximum gas capacity | 10*T | :white_check_mark: Through `MaxCapacityFactor` (default 10)
+| P | minimum target gas consumption per second | 1,000,000 | :x:
+| D | target gas consumption rate update constant | 2^25 | :x:
+| Q | target gas consumption rate update factor change limit | 2^15 | :x:
+| M | minimum gas price | 1 Wei | :white_check_mark:
+| K | initial gas price update factor | ~87*T | :white_check_mark: Through `TimeToDouble` (default 60s)
+
+#### Adjustment to ACP-176 calculations for price discovery
+
+// TODO: MinGasPrice requires a binary search to find the correct value (similar to how we do for DesiredTargetExcess in Coreth)
+
+### Genesis Configuration
+
+There will be a new genesis chain configuration to set the parameters for the chain without requiring the ACP176FeeManager precompile to be activated. This will be similar to the existing fee configuration parameters in chain configuration. If there is no genesis configuration for the new fee parameters the default values for C-Chain will be used. This will look like the following:
+
+```json
+{
+  ...
+  "acp224Timestamp": ...,
+  "acp224FeeConfig": {
+    "targetGas": ...,
+    "minGasPrice": ...,
+    "maxCapacityFactor": ...,
+    "timeToDouble": ...,
+    "targetBlockRate": ...,
+    "minBlockGasCost": ...,
+    "maxBlockGasCost": ...,
+    "blockGasCostStep": ...
```

Comment:
should the precompile enforce any upper or lower bounds on these dynamic values?